### PR TITLE
Add MariaDB OpsReq and Autoscaler Docs

### PR DESCRIPTION
**Pull request by spectro30**

### CODE REVIEW

File: docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+$ kubectl get mariadbopsrequest -n demo
+Every 2.0s: kubectl get mariadbopsrequest -n demo
+NAME                     TYPE              STATUS       AGE
+mdops-vscale        VerticalScaling      Successful    3m56s
+```
+
+We can see from the above output that the `MariaDBOpsRequest` has succeeded. Now, we are going to verify from one of the Pod yaml whether the resources of the database has updated to meet up the desired state, Let's check,
+
+```bash
+$ kubectl get pod -n demo sample-mariadb -o json | jq '.spec.containers[].resources'
```
Comment (DRoyyy) [inline docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md:165]: $ kubectl get pod -n demo sample-mariadb-0 -o json | jq '.spec.containers[].resources'

File: docs/guides/mariadb/upgrading/cluster/index.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    name: sample-mariadb
+  upgrade:
+    targetVersion: "10.5.8"
+```
+
+Here,
+
+- `spec.databaseRef.name` specifies that we are performing operation on `sample-mariadb` MariaDB database.
+- `spec.type` specifies that we are going to perform `Upgrade` on our database.
+- `spec.upgrade.targetVersion` specifies the expected version of the database `4.0.5`.
```
Comment (DRoyyy) [inline docs/guides/mariadb/upgrading/cluster/index.md:114]: `spec.upgrade.targetVersion` specifies the expected version of the database `10.5.8`.
//as per the yaml

File: docs/guides/mariadb/volume-expansion/volume-expansion/index.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+```bash
+$ kubectl get storageclass
+NAME                  PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
+standard (default)    rancher.io/local-path   Delete          WaitForFirstConsumer   false                  69s
+topolvm-provisioner   topolvm.cybozu.com      Delete          WaitForFirstConsumer   true                   37s
+
+```
+
+We can see from the output the `topolvm-provisioner` storage class has `ALLOWVOLUMEEXPANSION` field as true. So, this storage class supports volume expansion. We can use it.
```
Comment (DRoyyy) [inline docs/guides/mariadb/volume-expansion/volume-expansion/index.md:57]: So, this storage class supports volume expansion. We can use it. You can install topolvm from [here](https://github.com/topolvm/topolvm)

File: docs/guides/mariadb/volume-expansion/volume-expansion/index.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+```
+
+#### Verify MariaDB volume expanded successfully
+
+If everything goes well, `KubeDB` Enterprise operator will update the volume size of `MariaDB` object and related `StatefulSets` and `Persistent Volumes`.
+
+Let's wait for `MariaDBOpsRequest` to be `Successful`.  Run the following command to watch `MariaDBOpsRequest` CR,
+
+```bash
+$ kubectl get mariadbopsrequest -n demo
```
Comment (DRoyyy) [inline docs/guides/mariadb/volume-expansion/volume-expansion/index.md:164]: $ watch kubectl get mariadbopsrequest -n demo
//as per the above description

File: docs/guides/mariadb/upgrading/cluster/index.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+```
+
+You can see from above, our `MariaDB` cluster database has been updated with the new version. So, the upgrade process is successfully completed.
+
+## Cleaning Up
+
+To clean up the Kubernetes resources created by this tutorial, run:
+
+```bash
+kubectl delete mg -n demo sample-mariadb
```
Comment (DRoyyy) [inline docs/guides/mariadb/upgrading/cluster/index.md:158]: $ kubectl delete mariadb -n demo sample-mariadb

File: docs/guides/mariadb/upgrading/cluster/index.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+kubectl delete mariadbopsrequest -n demo mdops-upgrade
```
Comment (DRoyyy) [inline docs/guides/mariadb/upgrading/cluster/index.md:159]: $ kubectl delete mariadbopsrequest -n demo mdops-upgrade

File: docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+```
+
+The above output verifies that we have successfully scaled up the resources of the MariaDB database.
+
+## Cleaning Up
+
+To clean up the Kubernetes resources created by this tutorial, run:
+
+```bash
+kubectl delete mariadb -n demo sample-mariadb
```
Comment (DRoyyy) [inline docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md:185]: $ kubectl delete mariadb -n demo sample-mariadb

File: docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+kubectl delete mariadbopsrequest -n demo mdops-vscale
```
Comment (DRoyyy) [inline docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md:186]: $ kubectl delete mariadbopsrequest -n demo mdops-vscale

File: docs/guides/mariadb/scaling/horizontal-scaling/cluster/index.md [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+```
+
+From all the above outputs we can see that the replicas of the cluster is `5`. That means we have successfully scaled down the replicas of the MariaDB replicaset.
+
+## Cleaning Up
+
+To clean up the Kubernetes resources created by this tutorial, run:
+
+```bash
+kubectl delete mariadb -n demo sample-mariadb
```
Comment (DRoyyy) [inline docs/guides/mariadb/scaling/horizontal-scaling/cluster/index.md:272]: $ kubectl delete mariadb -n demo sample-mariadb

File: docs/guides/mariadb/scaling/horizontal-scaling/cluster/index.md [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+kubectl delete mariadbopsrequest -n demo  mdops-scale-horizontal-up mdops-scale-horizontal-down
```
Comment (DRoyyy) [inline docs/guides/mariadb/scaling/horizontal-scaling/cluster/index.md:273]: $ kubectl delete mariadbopsrequest -n demo  mdops-scale-horizontal-up mdops-scale-horizontal-down

File: docs/guides/mariadb/volume-expansion/volume-expansion/index.md [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+```
+
+The above output verifies that we have successfully expanded the volume of the MariaDB database.
+
+## Cleaning Up
+
+To clean up the Kubernetes resources created by this tutorial, run:
+
+```bash
+kubectl delete mg -n demo sample-mariadb
```
Comment (DRoyyy) [inline docs/guides/mariadb/volume-expansion/volume-expansion/index.md:244]: $ kubectl delete mariadb -n demo sample-mariadb

File: docs/guides/mariadb/volume-expansion/volume-expansion/index.md [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+kubectl delete mariadbopsrequest -n demo md-online-volume-expansion
```
Comment (DRoyyy) [inline docs/guides/mariadb/volume-expansion/volume-expansion/index.md:245]: $ kubectl delete mariadbopsrequest -n demo md-online-volume-expansion

File: docs/guides/mariadb/reconfigure/cluster/index.md [hunk 13]
```diff
@@ -0,0 +1,391 @@
...
+```
+
+As we can see from the configuration has changed to its default value. So removal of existing custom configuration using `MariaDBOpsRequest` is successful.
+
+## Cleaning Up
+
+To clean up the Kubernetes resources created by this tutorial, run:
+
+```bash
+kubectl delete mariadb -n demo sample-mariadb
```
Comment (DRoyyy) [inline docs/guides/mariadb/reconfigure/cluster/index.md:389]: $ kubectl delete mariadb -n demo sample-mariadb

File: docs/guides/mariadb/reconfigure/cluster/index.md [hunk 14]
```diff
@@ -0,0 +1,391 @@
...
+kubectl delete mariadbopsrequest -n demo mdops-reconfigure-config mdops-reconfigure-remove
```
Comment (DRoyyy) [inline docs/guides/mariadb/reconfigure/cluster/index.md:390]: $ kubectl delete mariadbopsrequest -n demo mdops-reconfigure-config mdops-reconfigure-remove

File: docs/guides/mariadb/reconfigure-tls/cluster/index.md [hunk 15]
```diff
@@ -0,0 +1,575 @@
...
+```
+
+So, we can see from the above that, output that tls is disabled successfully.
+
+## Cleaning up
+
+To cleanup the Kubernetes resources created by this tutorial, run:
+
+```bash
+kubectl delete mariadb -n demo --all
```
Comment (DRoyyy) [inline docs/guides/mariadb/reconfigure-tls/cluster/index.md:571]: $ kubectl delete mariadb -n demo --all

File: docs/guides/mariadb/reconfigure-tls/cluster/index.md [hunk 16]
```diff
@@ -0,0 +1,575 @@
...
+kubectl delete issuer -n demo --all
```
Comment (DRoyyy) [inline docs/guides/mariadb/reconfigure-tls/cluster/index.md:572]: $ kubectl delete issuer -n demo --all

File: docs/guides/mariadb/reconfigure-tls/cluster/index.md [hunk 17]
```diff
@@ -0,0 +1,575 @@
...
+kubectl delete mariadbopsrequest -n demo --all
```
Comment (DRoyyy) [inline docs/guides/mariadb/reconfigure-tls/cluster/index.md:573]: $ kubectl delete mariadbopsrequest -n demo --all

File: docs/guides/mariadb/reconfigure-tls/cluster/index.md [hunk 18]
```diff
@@ -0,0 +1,575 @@
...
+kubectl delete ns demo
```
Comment (DRoyyy) [inline docs/guides/mariadb/reconfigure-tls/cluster/index.md:574]: $ kubectl delete ns demo

**Review by DRoyyy — CHANGES_REQUESTED** [review]

File: docs/guides/mariadb/autoscaler/compute/cluster/examples/sample-mariadb.yaml [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+kind: MariaDB
+metadata:
+  name: sample-mariadb
+  namespace: demo
+spec:
+  version: "10.5.8"
+  replicas: 3
+  storageType: Durable
+  storage:
+    storageClassName: "topolvm-provisioner"
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/compute/cluster/examples/sample-mariadb.yaml:11]: we use kind for local cluster setup where default storage class is standard which doesn't support pvc related metrics. So we are using 'topolvm-provisioner' storage-class

File: docs/guides/mariadb/autoscaler/compute/cluster/index.md [hunk 20]
```diff
@@ -0,0 +1,541 @@
...
+{
+  "limits": {
+    "cpu": "200m",
+    "memory": "300Mi",
+    "topolvm.cybozu.com/capacity": "1"
+  },
+  "requests": {
+    "cpu": "200m",
+    "memory": "300Mi",
+    "topolvm.cybozu.com/capacity": "1"
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/compute/cluster/index.md:109]: we use kind for local cluster setup where default storage class is standard which doesn't support pvc related metrics. So we are using 'topolvm-provisioner' storage-class

File: docs/guides/mariadb/autoscaler/compute/cluster/index.md [hunk 21]
```diff
@@ -0,0 +1,541 @@
...
+
+Let's check the Pod containers resources,
+
+```bash
+$ kubectl get pod -n demo sample-mariadb-0 -o json | jq '.spec.containers[].resources'
+{
+  "limits": {
+    "cpu": "200m",
+    "memory": "300Mi",
+    "topolvm.cybozu.com/capacity": "1"
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/compute/cluster/index.md:104]: last comment

File: docs/guides/mariadb/autoscaler/compute/cluster/index.md [hunk 22]
```diff
@@ -0,0 +1,541 @@
...
+kind: MariaDB
+metadata:
+  name: sample-mariadb
+  namespace: demo
+spec:
+  version: "10.5.8"
+  replicas: 3
+  storageType: Durable
+  storage:
+    storageClassName: "topolvm-provisioner"
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/compute/cluster/index.md:63]: last comment

File: docs/guides/mariadb/autoscaler/compute/cluster/index.md [hunk 23]
```diff
@@ -0,0 +1,541 @@
...
+  Target Ref:
+    API Version:  apps/v1
+    Kind:         StatefulSet
+    Name:         sample-mariadb
+  Update Policy:
+    Update Mode:  Off
+Status:
+  Conditions:
+    Last Transition Time:  2022-01-13T13:06:13Z
+    Status:                True
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/compute/cluster/index.md:284]: fixed

File: docs/guides/mariadb/autoscaler/compute/cluster/index.md [hunk 24]
```diff
@@ -0,0 +1,541 @@
...
+    Type:                  RecommendationProvided
+  Recommendation:
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/compute/cluster/index.md:286]: fixed

File: docs/guides/mariadb/autoscaler/storage/cluster/index.md [hunk 25]
```diff
@@ -0,0 +1,322 @@
...
+      usageThreshold: 20
+      scalingThreshold: 20
+      expansionMode: "Online"
+```
+
+Here,
+
+- `spec.databaseRef.name` specifies that we are performing vertical scaling operation on `sample-mariadb` database.
+- `spec.storage.mariadb.trigger` specifies that storage autoscaling is enabled for this database.
+- `spec.storage.mariadb.usageThreshold` specifies storage usage threshold, if storage usage exceeds `60%` then storage autoscaling will be triggered.
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/storage/cluster/index.md:150]: fixed

File: docs/guides/mariadb/autoscaler/storage/cluster/index.md [hunk 26]
```diff
@@ -0,0 +1,322 @@
...
+- `spec.storage.mariadb.scalingThreshold` specifies the scaling threshold. Storage will be scaled to `50%` of the current amount.
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/storage/cluster/index.md:151]: fixed

File: docs/guides/mariadb/autoscaler/storage/cluster/index.md [hunk 27]
```diff
@@ -0,0 +1,322 @@
...
+#### Storage Autoscaling is set up successfully
+
+Let's check that the `mariadbautoscaler` resource is created successfully,
+
+```bash
+$ kubectl get mariadbautoscaler -n demo
+NAME           AGE
+mdas-storage   33s
+
+$ kubectl describe mariadbautoscaler mdas-storage -n demo
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/storage/cluster/index.md:170]: fixed

File: docs/guides/mariadb/autoscaler/storage/cluster/index.md [hunk 28]
```diff
@@ -0,0 +1,322 @@
...
+  Resource Version:  24009
+  UID:               4f45a3b3-fc72-4d04-b52c-a770944311f6
+Spec:
+  Database Ref:
+    Name:  sample-mariadb
+  Storage:
+    Mariadb:
+      Scaling Threshold:  50
+      Trigger:            On
+      Usage Threshold:    60
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/storage/cluster/index.md:190]: fixed

File: docs/guides/mariadb/autoscaler/storage/cluster/index.md [hunk 29]
```diff
@@ -0,0 +1,322 @@
...
+  Managed Fields:
+    ...
+  Resource Version:  24009
+  UID:               4f45a3b3-fc72-4d04-b52c-a770944311f6
+Spec:
+  Database Ref:
+    Name:  sample-mariadb
+  Storage:
+    Mariadb:
+      Scaling Threshold:  50
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/storage/cluster/index.md:188]: fixed

File: docs/guides/mariadb/autoscaler/storage/cluster/index.md [hunk 30]
```diff
@@ -0,0 +1,322 @@
...
+root@sample-mariadb-0:/ dd if=/dev/zero of=/var/lib/mysql/file.img bs=500M count=1
+1+0 records in
+1+0 records out
+524288000 bytes (524 MB, 500 MiB) copied, 0.340877 s, 1.5 GB/s
+root@sample-mariadb-0:/ df -h /var/lib/mysql
+Filesystem                                         Size  Used Avail Use% Mounted on
+/dev/topolvm/57cd4330-784f-42c1-bf8e-e743241df164 1014M  857M  158M  85% /var/lib/mysql
+```
+
+So, from the above output we can see that the storage usage is 83%, which exceeded the `usageThreshold` 60%.
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/storage/cluster/index.md:214]: fixed

File: docs/guides/mariadb/autoscaler/storage/cluster/index.md [hunk 31]
```diff
@@ -0,0 +1,322 @@
...
+
+Let's check that the `mariadbautoscaler` resource is created successfully,
+
+```bash
+$ kubectl get mariadbautoscaler -n demo
+NAME           AGE
+mdas-storage   33s
+
+$ kubectl describe mariadbautoscaler mdas-storage -n demo
+Name:         mdas-storage
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/storage/cluster/index.md:171]: fixed

File: docs/guides/mariadb/concepts/opsrequest/index.md [hunk 32]
```diff
@@ -0,0 +1,351 @@
...
+- `VolumeExpansion`
+- `Reconfigure`
+- `ReconfigureTLS`
+- `Restart`
+
+> You can perform only one type of operation on a single `MariaDBOpsRequest` CR. For example, if you want to upgrade your database and scale up its replica then you have to create two separate `MariaDBOpsRequest`. At first, you have to create a `MariaDBOpsRequest` for upgrading. Once it is completed, then you can create another `MariaDBOpsRequest` for scaling. You should not create two `MariaDBOpsRequest` simultaneously.
+
+### spec.upgrade
+
+If you want to upgrade you MariaDB version, you have to specify the `spec.upgrade` section that specifies the desired version information. This field consists of the following sub-field:
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/concepts/opsrequest/index.md:238]: fixed

File: docs/guides/mariadb/reconfigure/standalone/index.md [hunk 33]
```diff
@@ -0,0 +1,390 @@
...
+  databaseRef:
+    name: sample-mariadb
+  configuration:   
+    configSecret:
+      name: new-md-configuration
+```
+
+Here,
+
+- `spec.databaseRef.name` specifies that we are reconfiguring `mops-reconfigure-standalone` database.
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/reconfigure/standalone/index.md:193]: fixed

File: docs/guides/mariadb/reconfigure/cluster/index.md [hunk 34]
```diff
@@ -0,0 +1,391 @@
...
+  databaseRef:
+    name: sample-mariadb
+  configuration:   
+    configSecret:
+      name: new-md-configuration
+```
+
+Here,
+
+- `spec.databaseRef.name` specifies that we are reconfiguring `mops-reconfigure-cluster` database.
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/reconfigure/cluster/index.md:195]: fixed

File: docs/guides/mariadb/reconfigure/standalone/index.md [hunk 35]
```diff
@@ -0,0 +1,390 @@
...
+  type: Reconfigure
+  databaseRef:
+    name: sample-mariadb
+  configuration:   
+    removeCustomConfig: true
+```
+
+Here,
+
+- `spec.databaseRef.name` specifies that we are reconfiguring `mops-reconfigure-standalone` database.
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/reconfigure/standalone/index.md:323]: fixed

File: docs/guides/mariadb/reconfigure-tls/cluster/index.md [hunk 36]
```diff
@@ -0,0 +1,575 @@
...
+$ kubectl get mariadb -n demo
+NAME             VERSION   STATUS   AGE
+sample-mariadb   10.5.8    Ready    9m17s
+```
+
+```bash
+$ kubectl get secrets -n demo mg-rs-auth -o jsonpath='{.data.\username}' | base64 -d
+root
+
+$ kubectl get secrets -n demo mg-rs-auth -o jsonpath='{.data.\password}' | base64 -d
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/reconfigure-tls/cluster/index.md:84]: fixed

File: docs/guides/mariadb/reconfigure-tls/cluster/index.md [hunk 37]
```diff
@@ -0,0 +1,575 @@
...
+As we can see from the above output, the certificate has been rotated successfully.
+
+## Update Certificate
+
+Now, we are going to update the server certificate.
+
+- Let's describe the server certificate `sample-mariadb-server-cert`
+```bash
+$ 
+kubectl describe certificate -n demo sample-mariadb-server-cert
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/reconfigure-tls/cluster/index.md:346]: fixed

File: docs/guides/mariadb/reconfigure-tls/cluster/index.md [hunk 38]
```diff
@@ -0,0 +1,575 @@
...
+  type: ReconfigureTLS
+  databaseRef:
+    name: sample-mariadb
+  tls:
+    remove: true
+```
+
+Here,
+
+- `spec.databaseRef.name` specifies that we are performing reconfigure TLS operation on `mg-rs` database.
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/reconfigure-tls/cluster/index.md:509]: fixed

File: docs/guides/mariadb/autoscaler/compute/cluster/index.md [hunk 39]
```diff
@@ -0,0 +1,541 @@
...
+```bash
+$ kubectl get mariadbopsrequest -n demo
+NAME                              TYPE              STATUS       AGE
+mdops-vpa-sample-mariadb-z43wc8   VerticalScaling   Progressing  11s
+```
+
+Let's wait for the ops request to become successful.
+
+```bash
+kubectl get mariadbopsrequest -n demo
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/autoscaler/compute/cluster/index.md:428]: fixed

File: docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md [hunk 40]
```diff
@@ -0,0 +1,187 @@
...
+```bash
+$ kubectl get mariadb -n demo
+NAME             VERSION    STATUS     AGE
+sample-mariadb    10.5.8     Ready     3m46s
+```
+
+Let's check the Pod containers resources,
+
+```bash
+$ kubectl get pod -n demo sample-mariadb -o json | jq '.spec.containers[].resources'
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md:91]: fixed

File: docs/guides/mariadb/scaling/horizontal-scaling/cluster/index.md [hunk 41]
```diff
@@ -0,0 +1,274 @@
...
+$ watch kubectl get mariadbopsrequest -n demo
+Every 2.0s: kubectl get mariadbopsrequest -n demo
+NAME                          TYPE                STATUS       AGE
+mops-hscale-down-replicaset   HorizontalScaling   Successful   2m32s
+```
+
+We can see from the above output that the `MariaDBOpsRequest` has succeeded. Now, we are going to verify the number of replicas this database has from the MariaDB object, number of pods the statefulset have,
+
+```bash
+$ kubectl get mariadb -n demo mg-replicaset -o json | jq '.spec.replicas'
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/scaling/horizontal-scaling/cluster/index.md:248]: fixed

File: docs/guides/mariadb/scaling/horizontal-scaling/cluster/index.md [hunk 42]
```diff
@@ -0,0 +1,274 @@
...
+apiVersion: ops.kubedb.com/v1alpha1
+kind: MariaDBOpsRequest
+metadata:
+  name: mdops-scale-horizontal-down
+  namespace: demo
+spec:
+  type: HorizontalScaling
+  databaseRef:
+    name: sample-mariadb
+  horizontalScaling
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/scaling/horizontal-scaling/cluster/index.md:215]: fixed

File: docs/guides/mariadb/scaling/horizontal-scaling/cluster/example/mdops-downscale.yaml [hunk 43]
```diff
@@ -0,0 +1,11 @@
+apiVersion: ops.kubedb.com/v1alpha1
+kind: MariaDBOpsRequest
+metadata:
+  name: mdops-scale-horizontal-down
+  namespace: demo
+spec:
+  type: HorizontalScaling
+  databaseRef:
+    name: sample-mariadb
+  horizontalScaling
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/scaling/horizontal-scaling/cluster/example/mdops-downscale.yaml:10]: fixed

File: docs/guides/mariadb/reconfigure/overview/index.md [hunk 44]
```diff
@@ -0,0 +1,56 @@
...
+
+The Reconfiguring MariaDB process consists of the following steps:
+
+1. At first, a user creates a `MariaDB` Custom Resource (CR).
+
+2. `KubeDB` Community operator watches the `MariaDB` CR.
+
+3. When the operator finds a `MariaDB` CR, it creates required number of `StatefulSets` and related necessary stuff like secrets, services, etc.
+
+4. Then, in order to reconfigure the various components (ie. ReplicaSet, Shard, ConfigServer, Mongos, etc.) of the `MariaDB` database the user creates a `MariaDBOpsRequest` CR with desired information.
```
[the remark this answers is not in the document]
Comment (spectro30) [reply docs/guides/mariadb/reconfigure/overview/index.md:44]: fixed

File: docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md (hunk 1 above)
Comment (spectro30) [reply docs/guides/mariadb/scaling/vertical-scaling/cluster/index.md:165]: fixed

File: docs/guides/mariadb/upgrading/cluster/index.md (hunk 2 above)
Comment (spectro30) [reply docs/guides/mariadb/upgrading/cluster/index.md:114]: fixed

File: docs/guides/mariadb/volume-expansion/volume-expansion/index.md (hunk 3 above)
Comment (spectro30) [reply docs/guides/mariadb/volume-expansion/volume-expansion/index.md:57]: fixed

**Review by DRoyyy — DISMISSED** [review]
LGTM

**Review by DRoyyy — DISMISSED** [review]

**Pull request merged**